src/docs/asciidoc/api-guide.adoc

```diff
@@ -608,3 +608,39 @@ methods can be invoked after the receive Flux corresponding to the last receive
 
 To enable micrometer metrics for the underlying Kafka Consumers and Producers, add a `MicrometerConsumerListener` to the `ReceiverOptions` or a `MicrometerProducerListener` to the `SenderOptions` respectively.
 
+=== Micrometer Observation
+
+To enable Micrometer observations for produced and consumed records, add an `ObservationRegistry` to the `SenderOptions` and `ReceiverOptions` using `withObservation()` API.
+A custom `KafkaSenderObservationConvention` (and `KafkaReceiverObservationConvention`) can also be set.
+See their default implementations in the `KafkaSenderObservation` and `KafkaReceiverObservation`, respectively.
+The `DefaultKafkaSenderObservationConvention` exposes two low-cardinality tags: `reactor.kafka.type = sender`  and `reactor.kafka.client.id` with the `ProducerConfig.CLIENT_ID_CONFIG` option or identity hash code of the `DefaultKafkaSender` instance prefixed with the `reactor-kafka-sender-`.
+The `DefaultKafkaReceiverObservationConvention` exposes two low-cardinality tags: `reactor.kafka.type = receiver`  and `reactor.kafka.client.id` with the `ConsumerConfig.CLIENT_ID_CONFIG` option or identity hash code of the `DefaultKafkaReceiver` instance prefixed with the `reactor-kafka-receiver-`.
+
+If a `PropagatingSenderTracingObservationHandler` is configured on the `ObservationRegistry`, the tracing information from the context around producer record is stored into its headers before publishing this record to Kafka topic.
+If a `PropagatingReceiverTracingObservationHandler` is configured on the `ObservationRegistry`, the tracing information from the mentioned Kafka record headers, is restored into the context on a receiver side with a child span.
+
+Because the reverse order nature of the Reactor context, an observation functionality on the `KafkaReceiver` is limited just to a single `trace` logging message for received record.
```

Comment:
```suggestion
Because the reverse order nature of the Reactor context, the observation functionality on the `KafkaReceiver` is limited just to a single `trace` logging message for each received record.
```